docs/excel/custom-functions-get-started.md

```diff
@@ -0,0 +1,112 @@
+
+This article includes tips, best practices, and Office Add-ins ecosystem information for new custom functions add-in developers.
+
+The following diagram illustrates the interaction between a custom function and the two main components involved in custom function add-ins, Excel and external services.
+
+:::image type="content" source="../images/custom-functions-add-in-components.png" alt-text="The custom functions add-in communicates with both Excel and an external service, but Excel and the external service don't communicate directly with each other.":::
+
+**Excel** allows you to integrate your own custom functions into the application and run them like built-in functions.
+
+The **custom functions add-in** defines the logic for your functions and how they interact with Excel and Office JavaScript APIs. To learn how to create a custom functions add-in, see the [custom functions tutorial](../tutorials/excel-tutorial-create-custom-functions.md).
+
+An **external service** is optional. It can give your add-in capabilities like importing data from outside the workbook. The custom functions add-in specifies how external data is incorporated into the workbook. To learn more, see [Receive and handle data with custom functions](custom-functions-web-reqs.md).
+
+To develop a custom functions add-in with high performance, it’s important to ensure that these three components operate in harmony.
+
+## Optimize custom functions recalculation efficiency
+
+In general, custom functions recalculation follows the established pattern of [recalculation in Excel](/office/client-developer/excel/excel-recalculation). When recalculation is triggered, Excel enters a three-stage process: construct a dependency tree, construct a calculation chain, and then recalculate the cells. To optimize recalculation efficiency in your add-in, consider the level of nesting within your custom functions, the Excel calculation mode, and the limitations of volatile functions.
+
+### Nesting in custom functions
+
+A custom function can accept another custom function as an argument, ensuring that any dependent values are updated during recalculation. The recalculation of the outer custom function depends on the result of the nested function, leading to increased time consumption with each additional nested function. Minimize the number of nested levels in your custom functions to improve recalculation efficiency. The following code snippets demonstrate two approaches that produce similar outputs. **Option 1** is more likely to reduce the nested levels for end users when adding values in the workbook compared to **Option 2**.
+
+#### Option 1: Less nesting
+
+```js
+    /**
+    * Returns the sum of input numbers.
+    * @customfunction
+    */
+    function Add(args: number[]): number {
+      let total = 0;
+      args.forEach(value => {
+        total += value;
+      });
+     
+      return total;
+    }
+```
+
+#### Option 2: More nesting
```

Comment:
If this is the "bad" example, might be better to start with this, explain why it causes extra calc, and then show the better example. right now it looks like you can use either option equally well.

Reply:
Thanks for the feedback! I didn't end up restructuring, but I did add more explanation about which option is recommended and why.